sirepo_bluesky/sirepo_detector.py

```diff
@@ -101,10 +124,22 @@ def update_parameters(self):
 
     def trigger(self):
         super().trigger()
-        datum_id = new_uid()
+
         date = datetime.datetime.now()
-        srw_file = Path('/tmp/data') / Path(date.strftime('%Y/%m/%d')) / \
-            Path('{}.dat'.format(datum_id))
+        file_name = new_uid()
+        self._resource_document, self._datum_factory, _ = compose_resource(
+            start={'uid': 'needed for compose_resource() but will be discarded'},
+            spec='srw',
+            root='/tmp/data',
+            resource_path=str(Path(date.strftime('%Y/%m/%d')) / Path('{}.dat'.format(file_name))),
```

Comment:
Is it expected to be a relative path? Unfortunately, there is no docstring for https://github.com/bluesky/event-model/blob/9d3651ef8a4d8e2c3f046524c92bd5ff0ceefa2d/event_model/__init__.py#L1598-L1616.

Reply:
`resource_path` is intended to be relative to `root_path`, with the idea that if the files ever move only `root_path` will  have to be changed.

Reply:
Good, thanks for the confirmation.